e2e-test/README.md

```diff
@@ -0,0 +1,116 @@
+# End-to-End Tests
+
+The end-to-end tests use [Puppeteer](https://github.com/puppeteer/puppeteer/) and [Jest](https://jestjs.io/en/) to connect to a running instance of the application and use it through a browser like a normal user would.
+
+## Installation
+```bash
+cd e2e-test
+yarn install
+```
+
+Puppeteer includes a bundled Chromium executable, but not all the various Chrome dependencies.
+This is not generally an issue in development environments where Chrome is already installed.
+Be aware of this when setting up CI or Docker images though.
+
+## Running Tests
+You will need a running instance of the app, both the `web` and `flask` components.
+Assuming the web app is running at `http://localhost:8080`:
+```bash
+# within "test"
+CLIENT_URL=http://localhost:8080 yarn run test
+```
+
+## Debugging Tests
+Frequently when writing tests, they will not work the first time.
+For browser based tests, it is very helpful to be able to see the state of the browser at the point of failure.
+Use this command to run the browser in headful mode and extend the Jest test timeout to 1 hour:
+```bash
+# within "test"
+CLIENT_URL=http://localhost:8080 yarn run test-debug
+```
+You can also include `await jestPuppeteer.debug();` at any point in the test to create a manual breakpoint.
+
+## Writing Tests
+
+### Behavior Driven Development (BDD)
+Jest encourages BDD testing syntax.
+All tests should be formatted roughly like this:
+```javascript
+describe('thing being tested', () => {
+  it('should behave like this', async () => {
+    // ... test steps ...
+  });
+  it('does the right thing', async () => {
+    // ... test steps ...
+  });
+});
+```
+
+### expect-puppeteer
```

Comment:
Can you remove everything from here down?

Reply:
Do I keep line 49 or no?

Reply:
ah sorry, I left a comment on the wrong line. I'd keep this section, just remove stuff referring to stuff that was in `jest-vuetify-xpaths`. So line 58 and onward should be removed.